lib/hunter/commands/hunt.rb

```diff
@@ -71,20 +71,16 @@ def run
 
           client = server.accept
           first = false
-          hostid, hostname, payload = client.read.unpack("Z*Z*Z*")
-
-          node = {
-            "hostname" => hostname,
-            "ip" => (client.peeraddr[2] || 'unknown'),
-            "payload" => payload
-          }.reject { |k,v| v.empty? }
+          hostid, hostname, payload, label, prefix, group = client.read.unpack("Z*Z*Z*Z*Z*Z*")
 
           node = Node.new(
             id: hostid,
             hostname: hostname,
             ip: (client.peeraddr[2] || 'unknown'),
             payload: payload,
-            groups: []
+            groups: [group],
+            label: label,
+            prefix: prefix
```

Comment:
It's assumed that nodes in the buffer list don't have a label, as that's something that's awarded to a node when it graduates to the parsed list. I think these would better belong in a `presets` attribute that can be accessed in the upcoming new-and-improved parsing menu.